frontendFile/src/main/java/com/virtuslab/gitmachete/frontend/file/MacheteCompletionContributor.java

```diff
@@ -0,0 +1,49 @@
+package com.virtuslab.gitmachete.frontend.file;
+
+import com.intellij.codeInsight.completion.CompletionContributor;
+import com.intellij.codeInsight.completion.CompletionParameters;
+import com.intellij.codeInsight.completion.CompletionResultSet;
+import com.intellij.codeInsight.completion.PlainPrefixMatcher;
+import com.intellij.codeInsight.completion.impl.CamelHumpMatcher;
+import com.intellij.codeInsight.lookup.LookupElementBuilder;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import com.intellij.ui.TextFieldWithAutoCompletionListProvider;
+import git4idea.repo.GitRepositoryManager;
+import io.vavr.collection.List;
+
+public class MacheteCompletionContributor extends CompletionContributor {
+
+  @Override
+  public void fillCompletionVariants(CompletionParameters parameters, CompletionResultSet result) {
+
+    PsiFile file = parameters.getOriginalFile();
+    Project project = file.getProject();
+
+    var gitRepository = List.ofAll(GitRepositoryManager.getInstance(project).getRepositories())
+        .find(r -> file.getVirtualFile().getPath().startsWith(r.getRoot().getPath()));
+
+    if (gitRepository.isEmpty()) {
+      return;
+    }
+
+    var lists = List.ofAll(gitRepository.get().getInfo().getLocalBranchesWithHashes().keySet())
+        .map(localBranch -> localBranch.getName());
+
+    result.stopHere();
+    int count = parameters.getInvocationCount();
+
+    String prefix = TextFieldWithAutoCompletionListProvider.getCompletionPrefix(parameters);
```

Comment:
Hmmm back to the problem with line getting UN-indented after the completion, I think I might see the cause:

![image](https://user-images.githubusercontent.com/3383210/83660993-d510a680-a5c5-11ea-8e6c-efd6c13a5704.png)

For some incomprehensible (?) reason, `com.intellij.ui.TextFieldWithAutoCompletionListProvider#getCompletionPrefix(java.lang.String, int)` doesn't
take tab into account as prefix boundary, only space&newline. And .git/machete in my local ~/machete-sandbox uses **tabs** for indentation :/

Reply:
So likely `getCompletionPrefix` would need to be rewritten directly within our codebase